### ISSUE TITLE
Add __repr__ to all public classes for debuggability

### ISSUE THREAD

**Issue by sachncs** [sachncs/gradient-regularized-newton-boosting-trees#36]
## Problem

No public class in `grnbt/` defines a `__repr__`. Inspecting a fitted model in IPython, Jupyter, or after a `print(model)` shows the default:

```
<grnbt.boosting.VanillaNewtonBoosting object at 0x7f9b2c0e5f10>
```

This makes debugging, logging, and notebook exploration frustrating — users cannot see the configured hyperparameters at a glance.

## Evidence

- `grep -rn "__repr__" grnbt/` — no matches.
- All public classes: `Loss`, `MSELoss`, `CharbonnierLoss`, `BinaryCrossEntropyLoss`, `CategoricalCrossEntropyLoss`, `NewtonTree`, `MultiClassNewtonTree`, `Node`, `MultiClassNode`, `BaseBoosting`, `VanillaNewtonBoosting`, `GradientRegularizedNewtonBoosting`, `MultiClassNewtonBoosting`, `History`.

## Root Cause

`__repr__` was not implemented. The codebase focuses on correctness and validation, not on debug ergonomics.

## Impact

- **DX**: users cannot quickly inspect a fitted model.
- **Logging**: any `logger.info(model)` produces useless output.
- **Notebook exploration**: harder to compare two configurations.

## Recommendation

Add `__repr__` to each public class. Example:

```python
class VanillaNewtonBoosting:
    def __repr__(self) -> str:
        return (
            f"VanillaNewtonBoosting(loss={self.loss!r}, n_estimators={self.n_estimators}, "
            f"learning_rate={self.learning_rate}, max_depth={self.max_depth}, "
            f"lam_base={self.lam_base}, fitted={self.F0 is not None})"
        )
```

For trees, include the configured `max_depth`, `min_samples_leaf`, `min_gain`, and `n_bins` (for histogram). For losses, include `n_classes` (for CCE).

## Acceptance Criteria

- [ ] All public classes in `grnbt/` define `__repr__`.
- [ ] `repr(model)` shows the configured values.
- [ ] A test asserts `repr` includes the class name and key hyperparameters.

## Verification

```python
from grnbt import VanillaNewtonBoosting, MSELoss
m = VanillaNewtonBoosting(loss=MSELoss(), n_estimators=50, max_depth=4, lam_base=0.1)
print(repr(m))
```

should produce something like:

```
VanillaNewtonBoosting(loss=MSELoss(), n_estimators=50, learning_rate=1.0, max_depth=4, lam_base=0.1, fitted=False)
```

## Dependencies

None.

## Risk

None — purely additive.

**Comment by sachncs** [comment]
Resolved in PR #40.
